src/phasorpy/cursors.py

```diff
@@ -0,0 +1,299 @@
+    - modulation: array_like
+        Radial component of polar coordinates.
+    - phase_range: array_like, shape (..., 2)
+        Angular range of the cursors in radians.
+        The start and end of the range must be in the last dimension.
+    - modulation_range: array_like, shape (..., 2)
+        Radial range of the cursors.
+        The start and end of the range must be in the last dimension.
+    - axis: int, optional
+        Axis along which the masks are returned. Default is 0.
+
+    Returns
+    -------
+    - masks: ndarray
+        Polar coordinates masked for each polar cursor.
+
+    Raises
+    ------
+    ValueError
+        The array shapes of `phase` and `modulation`, or `phase_range` and
+        `modulation_range` do not match.
+        The last dimension of `phase_range` and `modulation_range` is not 2.
+        The axis is out of bounds.
+
+    Example
+    -------
+    Create mask from a single polar cursor:
+
+    >>> mask_from_polar_cursor([5, 100], [0.2, 0.4], [50, 150], [0.2, 0.5])
+    array([False,  True])
+
+    Create masks for two polar cursors with different ranges:
+
+    >>> mask_from_polar_cursor(
+    ...     [5, 100], [0.2, 0.4], [[50, 150], [0, 270]], [[0.2, 0.5], [0, 0.3]]
+    ... )
+    array([[False,  True],
+           [ True, False]])
+
+    """
+    phase = numpy.asarray(phase)
+    modulation = numpy.asarray(modulation)
+    phase_range = numpy.atleast_1d(phase_range)
+    modulation_range = numpy.atleast_1d(modulation_range)
+
+    if phase.shape != modulation.shape:
+        raise ValueError(f'{phase.shape=} != {modulation.shape=}')
+    if phase_range.shape != modulation_range.shape:
+        raise ValueError(f'{phase_range.shape=} != {modulation_range.shape=}')
+    if phase_range.shape[-1] != 2:
+        raise ValueError(f'{phase_range.shape[-1]=} != 2')
+    # TODO: check if angles are between -pi and pi
+
+    if numpy.ndim(phase_range) > 1:
+        axes = tuple(range(1, numpy.ndim(phase) + 1))
+        phase = numpy.expand_dims(phase, axis=0)
+        modulation = numpy.expand_dims(modulation, axis=0)
+        phase_range = numpy.expand_dims(phase_range, axis=axes)
+        modulation_range = numpy.expand_dims(modulation_range, axis=axes)
+    phase_mask = (phase >= phase_range[..., 0]) & (
+        phase <= phase_range[..., 1]
+    )
+    modulation_mask = (modulation >= modulation_range[..., 0]) & (
+        modulation <= modulation_range[..., 1]
+    )
+    masks = phase_mask & modulation_mask
+    if numpy.ndim(phase_range) > 1:
+        max_axis = masks.ndim - 1
+        if not (-max_axis - 1 <= axis <= max_axis):
+            raise ValueError(
+                f'Invalid axis {axis=}.'
+                f'Must be between {-max_axis - 1} and {max_axis}.'
+            )
+        masks = numpy.moveaxis(masks, 0, axis)
+    return masks
+
+
+def pseudo_color(
+    mean: NDArray,
+    masks: NDArray,
+    /,
+    *,
+    colors: ArrayLike = CATEGORICAL,
```

Comment:
This does not render well with Sphinx.

Reply:
@cgohlke I modified the default value so that it doesn't expand all the values from `CATEGORICAL`, although I don't think this is the optimal solution. If you think there is a better alternative, please let us know.

Reply:
Does mypy accept `str` for `ArrayLike`? I would use `colors: ArrayLike | None = None`, then `if colors is None: colors = CATEGORICAL ...`.